lib/ckb/utils.rb

```diff
@@ -65,28 +65,30 @@ def self.sign_sighash_all_inputs(inputs, outputs, privkey, pubkeys)
       [inputs, witnesses]
     end
 
-    def self.pubkey_hash(pubkey)
+    def self.pubkey_blake160(pubkey)
       pubkey_bin = hex_to_bin(pubkey)
       hash_bin = CKB::Blake2b.digest(pubkey_bin)
-      bin_to_hex(hash_bin)
+      bin_to_hex(hash_bin[0...20])
     end
 
-    def self.generate_address(prefix, pubkey_hash)
-      pubkey_hash_bin = hex_to_bin(pubkey_hash)
-      Bech32.encode(prefix, "\x00\x00\x00\x00\x00\x02" + pubkey_hash_bin)
+    # payload = type(\x00) | bin-idx(\x00\x00\x00\x02) | pubkey blake160
+    def self.generate_address(prefix, pubkey_blake160)
+      pubkey_blake160_bin = hex_to_bin(pubkey_blake160)
+      payload = "\x00\x00\x00\x00\x02" + pubkey_blake160_bin
```

Comment:
If we agree on the doc here: https://github.com/nervosnetwork/ckb/wiki/Common-Address-Format, we should use 00000003

Reply:
Using new format "P2PH" instead